File: #2966.divide-array-into-arrays-with-max-difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        vector<vector<int>> v;
        std::sort(nums.begin(),nums.end());
        for (int i = 0; i < nums.size(); i+=3) {
            if (nums[i + 2] - nums[i] > k) {
                return {};
            }

            v.push_back({nums[i], nums[i + 1], nums[i + 2]});
        }
        return v;
    }
};

File: #2966.divide-array-into-arrays-with-max-difference.ts
function divideArray(nums: number[], k: number): number[][] {
    nums.sort((a, b) => a - b);
    const out = Array.from({length: nums.length / 3}, 
        (_, i) => nums.slice(i * 3, i * 3 + 3));
    return out.some(arr => (arr.at(-1) - arr.at(0)) > k) ? [] : out;
};
